File: backend/src/main/java/com/example/backend/Employee.java
package com.example.backend;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Entity
@Data
@Table(name = "Employees")
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "employee_number")
    private Long employeeNumber;

    private String password;

    @Column(name = "mobile_number", length = 13)
    private String mobileNumber;

    @Column(name = "email_id", length = 100)
    private String emailId;

    @Column(length = 10)
    private String gender;

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Family> familyMembers;
}

File: frontend/lib/models/employee.dart
class Employee {
  final int employeeNumber;
  final String password;
  final String emailId;
  final String gender;

  Employee({
    required this.employeeNumber,
    required this.password,
    required this.emailId,
    required this.gender,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      employeeNumber: json['employeeNumber'],
      password: json['password'],
      emailId: json['emailId'],
      gender: json['gender'],
    );
  }
}
